packages/cyberstorm/src/components/DropDown.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { ReactElement, ReactNode, useState } from "react";
+import styles from "./componentStyles/DropDown.module.css";
+import { Button } from "./Button";
+import { faChevronDown } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
+import { DropDownItemProps } from "./DropDownItem";
+
+type DropDownProps = {
+  icon?: ReactNode;
+  defaultOpen?: boolean;
+  options?: DropDownOptions;
+  colorScheme?: "default" | "defaultDark" | "primary";
+  defaultSelectedOptionKey?: number;
+};
+
+export type DropDownOption = {
+  label: string;
+  content?: ReactElement<DropDownItemProps>;
+};
+export type DropDownOptions = Map<number | string, DropDownOption>;
+
+export const DropDown: React.FC<DropDownProps> = (props) => {
+  const { icon, colorScheme, defaultOpen, options, defaultSelectedOptionKey } =
+    props;
+
+  let initialValue: DropDownOption | null =
+    options?.values().next().value ?? null;
+
+  if (
+    options &&
+    defaultSelectedOptionKey &&
+    options.get(defaultSelectedOptionKey)
+  ) {
+    initialValue = options.get(defaultSelectedOptionKey) ?? null;
+  }
+
+  const [value, setValue] = useState(initialValue);
+  const dropDownItemElements = options
+    ? mapDropDownData(options, colorScheme, setValue)
+    : null;
+
+  return (
+    <div className={styles.root}>
+      <DropdownMenu.Root defaultOpen={defaultOpen}>
+        <DropdownMenu.Trigger asChild>
+          <div>
+            <Button
+              colorScheme={colorScheme}
+              rightIcon={icon}
+              label={value?.label ?? ""}
```

Comment:
Took a second review to connect the dots here. This shouldn't be the label of the `value` (the `DropDownItem`), but the label of the `DropDown` itself, right? The label that was removed from the `DropDown`'s as per my earlier comment since it wasn't used for anything.

Reply:
![image](https://user-images.githubusercontent.com/104134747/211297633-df15f086-8e11-4a31-a243-484d9f580b30.png)
This label is this one^, so it must change when the selected value changes. Also this is not the same as dropdownitem's title 
![image](https://user-images.githubusercontent.com/104134747/211297915-4569a033-2640-482b-a73e-ce714964e5c8.png)
 (new vs. newest)